README.md

```diff
@@ -197,6 +228,16 @@ TODO:
 - Outline our solution for these, in terms of observable effects for users (not in terms of spec primitives).
 - Maybe compare/contrast with iframes. Is it helpful or confusing to talk about how portals are more like popups than iframes? (I.e. they are top-level browsing contexts in spec terms.)
 
+### Opt-in
+
+Because of the [restrictions](#other-restrictions-while-portaled) on portaled content, especially the [storage restrictions](#storage-access-blocking), most existing content is not prepared to be rendered in a portal. As such, content will need to opt-in to being portaled; any content that does not so opt-in will cause the portal to fail to load.
+
+TODO:
+
+- Actually describe the opt-in, once we decide what it is.
+- Explain relation to `X-Frame-Options` and CSP `frame-ancestors`. These opt-outs probably become unnecessary if we have an explicit opt-in?
```

Comment:
Very much TBD, IMO. Might content want to allow itself to be prerendered offscreen but not displayed in a <portal>? My instinct is that if the opt in is to the generic "restricted prerendering" mode then it doesn't subsume XFO/CSP. We could add an explicit opt-in to portal embedding to that, perhaps.

That said, I think (but haven't thought deeply) that without credentials you would think there's less risk than XFO/frame-ancestors have historically protected against, insofar as the rendering it shows you is likely no different than if the referrer had taken their own screenshot of the web page and embedded that. We could consider revisiting this. Maybe file an issue if you think we should reopen that?

Reply:
I'd planned to treat it as part of the prerendering opt-in discussions, but yeah, we can have a dedicated issue.